sdk/core/azure-core/src/main/java/com/azure/core/http/HttpClientOptions.java

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http;
+
+import com.azure.core.util.Configuration;
+
+import java.time.Duration;
+
+/**
+ * General configuration options for {@link HttpClient HttpClients}.TableEntity
+ */
+public class HttpClientOptions {
```

Comment:
Do we need a new type here? We already have `ClientOptions` which also is used for configuring client-specific settings.  There are pros and cons to using ClientOptions vs creating new HttpClientOptions.

## Using ClientOptions:
Pros:
- ClientOptions is already GA'd and is wired on to many client builders. So, adding a new configuration to ClientOptions will require no further changes to client builders in terms of API.
- User has to learn about just one `ClientOptions` type 

Cons:
- ClientOptions is used for both AMQP and HTTP clients. So, some of the configurations in HTTP client options may not apply for AMQP services


## Using HttpClientOptions:
Pros:
- Contains only configurations applicable to HTTP-based services

Cons:
- Requires updating client builders of all services to add a new option to set HttpClientOptions
- Users will see two setters: `clientOptions(ClientOptions clientOptions)` and `httpClientOptions(HttpClientOptions httpClientOptions)`